clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeChecker.java

```diff
@@ -441,4 +535,43 @@ private Result checkDistributors(Node node, int clusterStateVersion) {
         return allowSettingOfWantedState();
     }
 
+    private Set<Integer> groupsWithUserWantedStateNotUp() {
+        return clusterInfo.getAllNodeInfos().stream()
+                          .filter(sni -> !UP.equals(sni.getUserWantedState().getState()))
+                          .map(NodeInfo::getGroup)
+                          .filter(Objects::nonNull)
+                          .filter(Group::isLeafGroup)
```

Comment:
In a valid configuration nodes should always be configured as part of leaf groups, so I think this is always true. But it doesn't hurt to be explicit about the expectations of the node.